kai/server.py

```diff
@@ -51,6 +54,37 @@
     "data/jsonschema/",
 )
 
+DEMO_MODE = os.getenv("DEMO_MODE") == "true"
+
+
+@contextmanager
+def playback_if_demo_mode(model_id, application_name, filename):
+    """A context manager to conditionally use a VCR cassette when demo mode is enabled."""
+    if DEMO_MODE:
+        KAI_LOG.debug(
+            f"DEMO MODE - using cassette {application_name}/{model_id}/{filename}.yaml"
+        )
+        my_vcr = vcr.VCR(
+            cassette_library_dir=f"{os.path.dirname(__file__)}/data/vcr/{application_name}/{model_id}/",
+            record_mode="once",
+            match_on=[
+                "uri",
+                "method",
+                "scheme",
+                "host",
+                "port",
+                "path",
+                "query",
+                "headers",
+                "body",
+            ],
+            record_on_exception=False,
+        )
+        with my_vcr.use_cassette(f"{filename}.yaml", filter_headers=["authorization"]):
+            yield
+    else:
+        yield
+
 
 def load_config():
```

Comment:
`load_config`, `load_templates`, `load_template`, and  `generate_prompt` are unused, I think. It should be safe to delete

Reply:
:+1:. will leave that for a separate PR though